### DISCUSSION TITLE
Get values from database

### DISCUSSION THREAD

**Discussion by I-SER-I** [notion-dotnet/notion-sdk-net#131] [Q&A] [answered]
I want to get all values ​​from the database. But for some reason, i only see the names of the properties, and the values ​​are empty
![image](https://user-images.githubusercontent.com/50848091/134228433-22c5a561-572f-4e33-ab6e-23d31c93f6f7.png)

Example Table
![image](https://user-images.githubusercontent.com/50848091/134228726-7dcc28a8-d1a1-4f3b-bfac-3f4d7cd445c4.png)

**Comment by KoditkarVedant** [comment, marked as answer]
@I-SER-I you need to use the `Query` API on the database to get all values (i.e. rows - pages in the terms of Notion).

**Reply by I-SER-I** [reply]
Many thanks

**Reply by dtwarogpl** [reply]
@KoditkarVedant  I'm stuck in exactly same situation. Could you provide any code example?

**Reply by I-SER-I** [reply]
> @KoditkarVedant I'm stuck in exactly same situation. Could you provide any code example?

```cs
var queryParams = new DatabasesQueryParameters { Filter = null };
var pages = await client.Databases.QueryAsync(databaseId, queryParams);
```
[source](https://github.com/notion-dotnet/notion-sdk-net#querying-a-database)

`databaseId` u can get on `Databases` property

**Reply by dtwarogpl** [reply]
@I-SER-I  This is exactly what I have done: 

` 
         public async IAsyncEnumerable<string> DoSomething()
        {
            var client = NotionClientFactory.Create(new ClientOptions
            {
                AuthToken = AuthTokentoken
            });

            var queryParams = new DatabasesQueryParameters { Filter = null };

            var pages = await client.Databases.QueryAsync(dbKey, queryParams);
        
            foreach (var Page in pages.Results)
            {
              foreach (var property in Page.Properties.Where(x => x.Value.Type!=PropertyValueType.Title))
                {

                    if(property.Value is RelationPropertyValue relationProperty)
                    {
                        yield return string.Join(",", relationProperty.Relation.Select(x => x.Id.ToString()));
                    }
                       
                    yield return property.Key + " - " + property.Value.Id + " - " + property.Value.Type;
                                
                }
            }
            
        }`

When my Notion database looks like this:
![image](https://user-images.githubusercontent.com/25666930/211492604-adeb10d8-9be7-418e-9532-cb032485a5f9.png)
I should be able to see those related pages but they are not there:
![image](https://user-images.githubusercontent.com/25666930/211493134-898bf885-4238-4a73-b830-bfc99e240d93.png)
Marked filed should give me 'Relation1Page'

**Reply by dtwarogpl** [reply]
Ok, solved by myself. Maybe  this would be helpfull for anyone in future. In my case there were always empty relations. I had to enable my related table in Notion to use  same integration as parent.
